File: clinica/web/src/lib/services/admin.service.ts
import api from '../api';
import { UsuarioAplicacion, Medico, EmpleadoEmp2024, RolesPermisosData } from '../types/domain';

export interface AdminDashboardData {
    totalUsuarios: number;
    medicosActivos: number;
    pacientesActivos: number;
    ultimosUsuarios: UsuarioAplicacion[];
}

export const AdminService = {
    getDashboard: async (): Promise<AdminDashboardData> => {
        const response = await api.get<AdminDashboardData>('/admin/dashboard');
        return response.data;
    },

    getUsuarios: async (): Promise<UsuarioAplicacion[]> => {
        const response = await api.get<UsuarioAplicacion[]>('/admin/usuarios');
        return response.data;
    },

    crearUsuario: async (usuario: any): Promise<UsuarioAplicacion> => {
        const response = await api.post<UsuarioAplicacion>('/admin/usuarios', usuario);
        return response.data;
    },

    updateUsuario: async (id: number | string, usuario: Partial<UsuarioAplicacion>): Promise<UsuarioAplicacion> => {
        const response = await api.patch<UsuarioAplicacion>(`/admin/usuarios/${id}`, usuario);
        return response.data;
    },

    getMedicos: async (): Promise<Medico[]> => {
        const response = await api.get<Medico[]>('/admin/medicos');
        return response.data;
    },

    crearMedico: async (medico: any): Promise<Medico> => {
        const response = await api.post<Medico>('/admin/medicos', medico);
        return response.data;
    },

    getEmpleados: async (params?: { carnet?: string }): Promise<EmpleadoEmp2024[]> => {
        const response = await api.get<EmpleadoEmp2024[]>('/admin/empleados', { params });
        return response.data;
    },

    getReportesAtenciones: async (filters?: any): Promise<any[]> => {
        const response = await api.get<any[]>('/admin/reportes/atenciones', { params: filters });
        return response.data;
    },

    getRolesPermisos: async (): Promise<RolesPermisosData> => {
        const response = await api.get<RolesPermisosData>('/admin/roles-permisos');
        return response.data;
    },
};

File: clinica/mobile_app/lib/services/admin_service.dart
import '../core/api_client.dart';

class AdminService {
  final ApiClient _api = ApiClient();

  Future<Map<String, dynamic>> getDashboard() async {
    final r = await _api.dio.get('/admin/dashboard');
    return r.data;
  }

  Future<List<dynamic>> getUsuarios() async {
    final r = await _api.dio.get('/admin/usuarios');
    return r.data;
  }

  Future<Map<String, dynamic>> crearUsuario(Map<String, dynamic> data) async {
    final r = await _api.dio.post('/admin/usuarios', data: data);
    return r.data;
  }

  Future<List<dynamic>> getMedicos() async {
    final r = await _api.dio.get('/admin/medicos');
    return r.data;
  }

  Future<List<dynamic>> getEmpleados({String? carnet}) async {
    final r = await _api.dio.get(
      '/admin/empleados',
      queryParameters: carnet != null ? {'carnet': carnet} : null,
    );
    return r.data;
  }

  Future<Map<String, dynamic>> getRolesPermisos() async {
    final r = await _api.dio.get('/admin/roles-permisos');
    return r.data;
  }

  Future<List<dynamic>> getReportesAtenciones() async {
    final r = await _api.dio.get('/admin/reportes/atenciones');
    return r.data;
  }
}
